### DISCUSSION TITLE
Support for .NET 7.0 (and beyond)

### DISCUSSION THREAD

**Discussion by codefactors** [QuestPDF/QuestPDF#586] [General]
Hi - apologies if this is documented elsewhere but I am using .NET 7.0 for new projects where I don't have to publish the source; the plan being to move to the LTS .NET 8.0 when it ships.  But it seems that QuestPDF does not support .NET 7.0, at least for the previewer.

Is there a plan to address this?  Many thanks in advance (Steve).

[I should add that I would be happy to do a PR if that helps.]

**Comment by MarcinZiabek** [comment]
QuestPDF fully supports .NET 7 and every release beyond (because .NET releases are backward compatible).

To run the QuestPDF Previewer application, please install the .NET 6 runtime on your PC. The Previewer application can run on a different runtime than your project without any problem.

**Reply by codefactors** [reply]
Apologies @MarcinZiabek for not responding sooner on this.  The issue I was facing is that when I started my .NET 7 application and called `ShowInPreviewer()`, I got an error about compatibility, and assumed it was to do with .NET versions.

Coming back to this, I subsequently got this message:

![quest_previewer_is_not_compatible](https://github.com/QuestPDF/QuestPDF/assets/132352961/e6efe515-829f-42d4-a54a-ba7ed1a0719c)

When I followed the instructions to update the previewer, I got the following:

```cmd
C:\Users\Codefactors>dotnet tool update --global QuestPDF.Previewer --version 2023.5
The requested version 2023.5.0 is lower than existing version 2023.6.0.
```

It turned out the issue was that I had the latest version of the previewer but an older version of the library, as your warning suggested.  It might nevertheless be worth updating the error message regarding the previewer version to make it more generic.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
